internal/common/utils/DynamicMockTranslator.go

```diff
@@ -0,0 +1,67 @@
+package utils
+
+import (
+	"context"
+	"strings"
+
+	"github.com/RedHatInsights/tenant-utils/pkg/tenantid"
+)
+
+type dynamicMockTranslator struct {
+}
+
+func NewDynamicMockTranslator() tenantid.Translator {
+	return &dynamicMockTranslator{}
+}
+
+func (this *dynamicMockTranslator) EANsToOrgIDs(ctx context.Context, eans []string) (results []tenantid.TranslationResult, err error) {
+	results = make([]tenantid.TranslationResult, len(eans))
+
+	for _, ean := range eans {
+		newEAN := new(string)
+		*newEAN = ean
+		r := tenantid.TranslationResult{
+			OrgID: ean + "-test",
+			EAN:   newEAN,
+			Err:   nil,
+		}
+		results = append(results, r)
+	}
+
+	return results, nil
+}
+
+func (this *dynamicMockTranslator) OrgIDsToEANs(ctx context.Context, orgIDs []string) (results []tenantid.TranslationResult, err error) {
+	results = make([]tenantid.TranslationResult, len(orgIDs))
+
+	for _, orgID := range orgIDs {
+		newOrgID := new(string)
+		*newOrgID = orgID
+
+		if strings.HasSuffix(orgID, "-test") {
+			ean := strings.TrimSuffix(orgID, "-test")
```

Comment:
@CodyWMitchell there's quite a lot of code duplication happening here. Can you try reducing that? Ideally we define the transformation once and push the rest of the code back to tenant-utils so that other apps can reuse the mock the same way (provided they provide their own transformation functions)